plugin/pkg/scheduler/schedulercache/cache_test.go

```diff
@@ -0,0 +1,484 @@
+			{pod: testPods[1], assumedTime: now.Add(3 * ttl / 2)},
+		},
+		cleanupTime: now.Add(2 * ttl),
+		wNodeInfo: &NodeInfo{
+			requestedResource: &Resource{
+				MilliCPU: 200,
+				Memory:   1024,
+			},
+			nonzeroRequest: &Resource{
+				MilliCPU: 200,
+				Memory:   1024,
+			},
+			pods: []*api.Pod{testPods[1]},
+		},
+	}}
+
+	for i, tt := range tests {
+		cache := newSchedulerCache(ttl, time.Second, nil)
+
+		for _, pod := range tt.pods {
+			if err := cache.assumePodIfBindSucceed(pod.pod, alwaysTrue, pod.assumedTime); err != nil {
+				t.Fatalf("assumePod failed: %v", err)
+			}
+		}
+		// pods that have assumedTime + ttl < cleanupTime will get expired and removed
+		cache.cleanupAssumedPods(tt.cleanupTime)
+		n := cache.nodes[nodeName]
+		if !reflect.DeepEqual(n, tt.wNodeInfo) {
+			t.Errorf("#%d: node info get=%s, want=%s", i, n, tt.wNodeInfo)
+		}
+	}
+}
+
+// TestAddPodWillConfirm tests that a pod being Add()ed will be confirmed if assumed.
+// The pod info should still exist after manually expiring unconfirmed pods.
+func TestAddPodWillConfirm(t *testing.T) {
+	nodeName := "node"
+	now := time.Now()
+	ttl := 10 * time.Second
+
+	testPods := []*api.Pod{
+		makeBasePod(nodeName, "test-1", "100m", "500", []api.ContainerPort{{HostPort: 80}}),
+		makeBasePod(nodeName, "test-2", "200m", "1Ki", []api.ContainerPort{{HostPort: 8080}}),
+	}
+	tests := []struct {
+		// 0, 1: pod to assume
+		// 2: pod to add
+		pods []*api.Pod
+
+		wNodeInfo *NodeInfo
+	}{{ // two pod were assumed at same time. But first one is called Add() and gets confirmed.
+		pods: []*api.Pod{testPods[0], testPods[1], testPods[0]},
+		wNodeInfo: &NodeInfo{
+			requestedResource: &Resource{
+				MilliCPU: 100,
+				Memory:   500,
+			},
+			nonzeroRequest: &Resource{
+				MilliCPU: 100,
+				Memory:   500,
+			},
+			pods: []*api.Pod{testPods[0]},
+		},
+	}}
+
+	for i, tt := range tests {
+		podToAssume1 := tt.pods[0]
+		podToAssume2 := tt.pods[1]
+		podToAdd := tt.pods[2]
+		cache := newSchedulerCache(ttl, time.Second, nil)
+		if err := cache.assumePodIfBindSucceed(podToAssume1, alwaysTrue, now); err != nil {
+			t.Fatalf("assumePod failed: %v", err)
+		}
+		if err := cache.assumePodIfBindSucceed(podToAssume2, alwaysTrue, now); err != nil {
+			t.Fatalf("assumePod failed: %v", err)
+		}
+		if err := cache.AddPod(podToAdd); err != nil {
+			t.Fatalf("AddPod failed: %v", err)
+		}
+		cache.cleanupAssumedPods(now.Add(2 * ttl))
+		// check after expiration. confirmed pods shouldn't be expired.
+		n := cache.nodes[nodeName]
+		if !reflect.DeepEqual(n, tt.wNodeInfo) {
+			t.Errorf("#%d: node info get=%s, want=%s", i, n, tt.wNodeInfo)
+		}
+	}
+}
+
+// TestAddPodAfterExpiration tests that a pod being Add()ed will be added back if expired.
+func TestAddPodAfterExpiration(t *testing.T) {
+	nodeName := "node"
+	ttl := 10 * time.Second
+	basePod := makeBasePod(nodeName, "test", "100m", "500", []api.ContainerPort{{HostPort: 80}})
+	tests := []struct {
```

Comment:
Please remove this struct here.Since you are testing a single specific scenario here, we don't want to have a loop for it here (as someone may want to add another testcase in the future, which basically doesn't make sense for this test).